### DISCUSSION TITLE
NGINX reverse proxy conf with a specific "location" directive

### DISCUSSION THREAD

**Discussion by WesTale** [requarks/wiki#4656] [Help / Questions] [answered]
Hi Guys,
i have a problem to config NGINX for handle Wiki.js using a specific location:

if i use:
```
location / {
    proxy_pass         "http://xxxxxxx:xxx/";
    proxy_set_header   Host               $host:$server_port;
    proxy_set_header   Upgrade            $http_upgrade;
    proxy_set_header   Connection         $http_connection;
    proxy_set_header   X-Real-IP          $remote_addr;
    proxy_set_header   X-Forwarded-For    $proxy_add_x_forwarded_for;
    proxy_set_header   X-Forwarded-Proto  $scheme;
}
```

All works like a charm, but on my NGINX i have "X" website to reverse proxy, so i cannot let this type of config and i have to use a different location, like:

```
location /wikijs {
    proxy_pass         "http://xxxxxxx:xxx/";
    proxy_set_header   Host               $host:$server_port;
    proxy_set_header   Upgrade            $http_upgrade;
    proxy_set_header   Connection         $http_connection;
    proxy_set_header   X-Real-IP          $remote_addr;
    proxy_set_header   X-Forwarded-For    $proxy_add_x_forwarded_for;
    proxy_set_header   X-Forwarded-Proto  $scheme;
}
```

but it seems i receive a redirect type problem
![image](https://user-images.githubusercontent.com/85457164/140045322-04bfdf85-b62d-4d55-a92a-9109ec9d8e32.png)

this is first time that happen, i have 4 other website that i manage with the same conf like, but i didn't have redirect problems.

any help it will be very appreciated :)

**Comment by ieaster1** [comment, marked as answer]
You need to configure a server block for your wiki instance to then access it as wikijs.domain.com.

```
 server {
     server_name wikijs.domain.com;
     access_log /var/log/nginx/wiki.access.log;
     error_log /var/log/nginx/wiki.error.log;

     location / {
         proxy_pass         "http://xxxxxxx:xxx/";
         proxy_set_header   Host               $host:$server_port;
         proxy_set_header   Upgrade            $http_upgrade;
         proxy_set_header   Connection         $http_connection;
         proxy_set_header   X-Real-IP          $remote_addr;
         proxy_set_header   X-Forwarded-For    $proxy_add_x_forwarded_for;
         proxy_set_header   X-Forwarded-Proto  $scheme;
     }
}
 ```

**Reply by WesTale** [reply]
So, as i understood by your reply, is not possible to "play" with the location parameter together with rewrite rule, you are constrained to create a dedicated server config for a wiki.js instance, right?

**Reply by ieaster1** [reply]
Correct.  The app doesn't work within a subfolder installation.

Ref: https://feedback.js.wiki/wiki/p/support-hosting-wikijs-at-subpath

**Reply by WesTale** [reply]
Many thanks for your support @ieaster1!!! At least now i know why isn't working.
